### PR TITLE
Add MD2 Hashing

### PR DESCRIPTION
**Pull request by Kalkwst**
Added a new algorithms category, cryptographic hashing.
This PR introduces the MD2 hash algorithm.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

**Comment by Kalkwst** [comment]
I have checked the Codecov report and noticed that it says the `Update` method is not tested. However, this method is called by the `Digest` method, which is tested and verified to generate correct MD2 hashes. You can compare the results with online MD2 calculators. Therefore, I think the `Update` method is already tested through the `Digest` method, and it works as expected.

If we want to test the Update method directly, we would have to change its visibility to public. This would also mean creating another public method that does the rest of the work that the `Digest` method does now. Then we would have to wrap these two methods in the Digest method for simplicity. But I think this would make the code harder to understand since we would introduce two new methods to test one.

File: Algorithms.Tests/Crypto/Digests/Md2DigestTests.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        "32EC01EC4A6DAC72C0AB96FB34C0B5D1",
+        "DA853B0D3F88D99B30283A69E6DED6BB",
+        "AB4F496BFB2A530B219FF33031FE06B0",
+        "4E8DDFF3650292AB5A4108C3AA47940B",
+        "DA33DEF2A42DF13975352846C30338CD",
+        "D5976F79D83D3A0DC9806C3C66F3EFD8",
+    };
+
+    [Test]
+    public void Digest_ReturnsCorrectValue()
```
Comment (siriak) [inline Algorithms.Tests/Crypto/Digests/Md2DigestTests.cs:36]: Please use `TestCase` attribute to supply inputs

Comment (Kalkwst) [reply Algorithms.Tests/Crypto/Digests/Md2DigestTests.cs]: I really think that there is an issue with how codecov checks the code at this point.

First there is the issue where lines 202-204 were not covered.

The condition on the `while` loop `(mBufferOffset != 0) && (length > 0)` was not met by the tests I've written.

This condition checks if there is any data left in the buffer (`mBufferOffset != 0`) and if there is any data left to process (`length = 0`). If both conditions are true, it processes the remaining data in the buffer before moving on to the next chunk of data.

To cover these lines, I created a test where `mBufferOffset` is not zero and `length` is greater than zero at the same time. This can be achieved by providing an input that is not a multiple of 16 (the size of the buffer), because the `Update` method processes data in chunks of 16 bytes.

For that reason, I added the following test case

```csharp
[TestCase("123456789012345678901234567890123456789012345678901234567890123456789012345678901", "ExpectedHash")]
```

The string has a length of 81, which is not a multiple of 16. Therefore, after processing 5 full chunks of 16 bytes, there will be a remainder of 1 byte in the buffer, which should trigger the `while((mBufferedOffset != 0) && (length > 0)) loop in the `Update` method.

**Review by siriak — APPROVED** [review]
Looks good, thanks!
